hack/format-scripts.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+[[ -n "${DEBUG:-}" ]] && set -x
+
+gopath="$(go env GOPATH)"
+
+# install shfmt that ensures consistent formal in scripts
+if ! [[ -x "${gopath}/bin/shfmt" ]]; then
+  echo >&2 'Installing shfmt'
+  curl --silent --fail --location https://github.com/mvdan/sh/releases/download/v2.6.4/shfmt_v2.6.4_linux_amd64 >"${gopath}/bin/shfmt"
+  chmod u+x "${gopath}/bin/shfmt"
+fi
+
+"$gopath/bin/shfmt" -w -i=2 hack
```

Comment:
I was thinking this PR would also have a `hack/validate-scripts-formatting.sh` that checks if this command prints any diff and if so it would print the diff to have the committer fix the errors by telling user to run `shfmt -w -i=2 hack/`. 

Should we do that in another PR?
If we don't fail the build, people won't realize they should be running this script.

Reply:
I agree so I added into lint 
https://github.com/kubernetes-sigs/krew/pull/331/files#diff-c8cc35391f3940f8aa5f58aed0516ca1R57

However, I can rename this file (that just formats all files) as your suggestion (`validate-script-formatting.sh` - note I dropped `s`, `script` instead of `scripts`)